scrapy/responsetypes.py

```diff
@@ -5,8 +5,20 @@
 from mimetypes import MimeTypes
 from pkgutil import get_data
 from io import StringIO
+from urllib.parse import urlparse
+from warnings import warn
 
-from scrapy.http import Response
+from xtractmime import RESOURCE_HEADER_BUFFER_LENGTH, extract_mime
+from xtractmime._utils import contains_binary
```

Comment:
Is it OK to import a private module here?

Reply:
I am not sure about this. Maybe we can move the `contains_binary` function to `__init__.py` of `xtractmime`. @Gallaecio ?

Reply:
:+1: 

I also suggest renaming it as `is_binary_data`, to be consistent with https://mimesniff.spec.whatwg.org/#binary-data-byte